### ISSUE TITLE
Very sensitive swiping in portrait mode

### ISSUE THREAD

**Issue by eNBeWe** [EventFahrplan/EventFahrplan#22]
When using the app in portrait orientation the swiping gesture to move to the next/previous room is very slow. This leads to usually swiping too fast and switching all the way to the first/last room.

System:
Nexus 5X
LineageOS 14.1, Android 7.1.2

Version:
1.33.0

**Comment by Noxmiles** [comment]
Yes, this is a Problem. If you don't know it, it seems Like there are only 2 rooms.

**Comment by johnjohndoe** [comment]
Thank you for reporting the issue. At the moment I cannot reproduce it - the problem does not occur on the device (Nexus 5, Android 6.0.1) I am using. I would love to try adjusting the `SWIPE_MIN_DISTANCE` and `SWIPE_THRESHOLD_VELOCITY` - but I cannot do that without the device. If you can compile the app yourself and try different values I would be interested in the values you come up with. The relevant constants can be found in [`HorizontalSnapScrollView.java`](https://github.com/EventFahrplan/EventFahrplan/blob/v.1.33.1/app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollView.java#L34-L36).
Other then that I can offer that we met at 34C3 and try it out together.

**Comment by eNBeWe** [comment]
I didn't find direct contact Info for you @johnjohndoe You can reach me at DECT 2311, so we can have a look together.

**Comment by johnjohndoe** [comment]
Can we meet around 17:30 at c-base assembly? I will try to call you.

**Comment by eNBeWe** [comment]
A follow-up after our recent analysis: What do you think about splitting the function of the two constants:
Velocity: Select between fling or touch logic
Distance: Number of swiped columns if flinging
?

I would see if I can get a toolchain running and try that out if you like the idea.

**Comment by johnjohndoe** [comment]
Sounds reasonable to investigate. I would certainly clean up the code before (variable names, formatting).

**Comment by johnjohndoe** [comment]
@eNBeWe I did a basic cleanup of the code in `HorizontalSnapScrollView` [here](https://github.com/EventFahrplan/EventFahrplan/pull/38) - it would be nice if you could review it. Feel free to comment - the code changes are not final.

**Comment by johnjohndoe** [comment]
@eNBeWe Can you please add the Android version you are running on your Nexus 5X above? I tried a Nexus 6, Android 5.1.1 but could **not** reproduce the swiping behavior.

**Comment by eNBeWe** [comment]
I am using LineageOS 14.1, resulting in Android 7.1.2

**Comment by johnjohndoe** [comment]
@Noxmiles Can you please post the specs of your device?

**Comment by johnjohndoe** [comment]
> ... I did a basic cleanup of the code in `HorizontalSnapScrollView` ...

The cleanup branch has been merged here: 6753a52de1e4d482bac02e907503058347764194

**Comment by Noxmiles** [comment]
@johnjohndoe Moto Z Play. Android 7.1.1. 1920 x 1080 @ 5,5" 16:9 (~400ppi). I could make a video of the problem if you want.

**Comment by johnjohndoe** [comment]
@Noxmiles Thank you. The video is not needed since I can in general reproduce the behavior. I just have no access to the relevant devices myself. I update the list above to include your device.

**Comment by johnjohndoe** [comment]
Maybe of interest: [What's New in ConstraintLayout](https://www.youtube.com/watch?v=29gLA90m6Gk&t=18m09s) mentions that with `MotionLayout` it is possible to track "your velocity of your finger and matches the velocity of the object you tag to it".

**Comment by isiko** [comment]
![screen-20220423-004047_2](https://user-images.githubusercontent.com/44641688/164814241-9a1d91e4-acc4-42ee-9de6-84dcc1293844.gif)

Here's a GIF of the Behavior I mentioned in #479. You can see that I'm flinging to the left, but after watching the GIF a few times, I think it is actually overshooting and then moving back. 

A possible solution to this would be to switch from calculating the target column when starting the fling, to just snapping to the next column when the velocity is lower than a certain value, although I wouldn't know how to integrate that with the Android API.

**Comment by johnjohndoe** [comment]
@isiko It is very hard to see when the overshot happens because the GIF replays all the time. Could you instead upload a video (which stops automatically or can be stopped)?

**Comment by isiko** [comment]
> @isiko It is very hard to see when the overshot happens because the GIF replays all the time. Could you instead upload a video (which stops automatically or can be stopped)?

Updated the Video 👍

**Comment by johnjohndoe** [comment]
Thank you. - I downloaded the video to playback frame by frame. What I see is:

1. Scrolls fast rightwards to room "Stage" which is the ~32. column
2. Scrolls leftwards to room "Eliza" which is the 5. column (4 columns have been scrolled in effect)
